easybuild/easyblocks/q/quantumespresso.py

```diff
@@ -25,816 +25,384 @@
-        """Add GIPAW support to the build."""
-        if self.cfg['with_gipaw']:
-            self.cfg.update('buildopts', 'gipaw', allow_duplicate=False)
+    def _add_hdf5(self):
+        """Enable HDF5 for Quantum ESPRESSO."""
+        if get_software_root('HDF5'):
+            self.cfg.update('configopts', '-DQE_ENABLE_HDF5=ON')
         else:
-            if 'gipaw' in self.cfg['buildopts']:
-                self.cfg['buildopts'] = self.cfg['buildopts'].replace('gipaw', '')
+            self.cfg.update('configopts', '-DQE_ENABLE_HDF5=OFF')
 
-    def _add_wannier90(self):
-        """Add Wannier90 support to the build."""
-        if self.cfg['with_wannier90']:
-            self.cfg.update('buildopts', 'w90', allow_duplicate=False)
+    def _add_libxc(self):
+        """Enable LibXC for Quantum ESPRESSO."""
+        if get_software_root('libxc'):
+            self.cfg.update('configopts', '-DQE_ENABLE_LIBXC=ON')
         else:
-            if 'w90' in self.cfg['buildopts']:
-                self.cfg['buildopts'] = self.cfg['buildopts'].replace('w90', '')
-
-    def _adjust_compiler_flags(self, comp_fam):
-        """Adjust compiler flags based on the compiler family and code version."""
-        if comp_fam == toolchain.INTELCOMP:
-            if LooseVersion("6.0") <= LooseVersion(self.version) <= LooseVersion("6.4"):
-                i_mpi_cc = os.getenv('I_MPI_CC', '')
-                if i_mpi_cc == 'icx':
-                    env.setvar('I_MPI_CC', 'icc')  # Needed as clib/qmmm_aux.c using <math.h> implicitly
-        elif comp_fam == toolchain.GCC:
-            pass
+            self.cfg.update('configopts', '-DQE_ENABLE_LIBXC=OFF')
 
-    def configure_step(self):
-        """Custom configuration procedure for Quantum ESPRESSO."""
+    def _add_elpa(self):
+        """Enable ELPA for Quantum ESPRESSO."""
+        if get_software_root('ELPA'):
+            if not self.cfg.get('with_scalapack', False):
+                raise EasyBuildError('ELPA support requires ScaLAPACK')
+            if LooseVersion(self.version) == LooseVersion('7.3') and self.toolchain.options.get('openmp', False):
+                raise EasyBuildError('QE 7.3 with cmake does not support ELPA with OpenMP')
+            self.cfg.update('configopts', '-DQE_ENABLE_ELPA=ON')
+        else:
+            self.cfg.update('configopts', '-DQE_ENABLE_ELPA=OFF')
 
-        if LooseVersion(self.version) >= LooseVersion("7.3.1"):
-            raise EasyBuildError(
-                "QuantumESPRESSO 7.3.1 and later are not supported with the this easyblock (ConfigureMake), " +
-                "use the EB_QuantumESPRESSOcmake (CMakeMake) easyblock instead."
-                )
+    def _add_gipaw(self):
+        """Enable GIPAW for Quantum ESPRESSO."""
+        if LooseVersion(self.version) == LooseVersion('7.3.1'):
+            # See issue: https://github.com/dceresoli/qe-gipaw/issues/19
+            raise EasyBuildError('GIPAW will fail to compile in QE 7.3.1')
```

Comment:
```suggestion
        if LooseVersion(self.version) == LooseVersion('7.3.1'):
            # See issue: https://github.com/dceresoli/qe-gipaw/issues/19
            raise EasyBuildError('GIPAW will fail to compile in QE 7.3.1')
```
No longer necessary due to the change in https://github.com/easybuilders/easybuild-easyconfigs/commit/a563666c80665dcb70cbf748bf5f487788db3dc0